OpenImis.ePayment/Escape/Payment/Data/GepgUtility.cs

```diff
@@ -158,42 +156,25 @@ public String CreateBill(IConfiguration Configuration, string OfficerCode, strin
                 BillTrxInf = billTrxInfs
             };
 
-            XmlSerializer xs = null;
-            XmlSerializerNamespaces ns = null;
-            XmlWriterSettings settings = null;
-            String outString = String.Empty;
-
-            XmlWriter xw = null;
-
-            try
-            {
-                settings = new XmlWriterSettings();
-                settings.OmitXmlDeclaration = true;
-                //settings.Indent = true;
-                ns = new XmlSerializerNamespaces();
-                ns.Add("", "");
-
-                StringBuilder sb = new StringBuilder();
-                xs = new XmlSerializer(typeof(gepgBillSubReq));
-
-                xw = XmlWriter.Create(sb, settings);
+            var settings = new XmlWriterSettings() { OmitXmlDeclaration = true };
+            var ns = new XmlSerializerNamespaces();
+            ns.Add("", "");
+            var sb = new StringBuilder();
+            var xs = new XmlSerializer(typeof(gepgBillSubReq));
 
-                xs.Serialize(xw, newBill, ns);
-                xw.Flush();
-                outString = sb.ToString();
-            }
-            catch (Exception ex)
-            {
-                Console.WriteLine(ex.ToString());
-            }
-            finally
+            using (var xw = XmlWriter.Create(sb, settings))
             {
-                if (xw != null)
+                try
                 {
-                    xw.Close();
+                    xs.Serialize(xw, newBill, ns);
+                    xw.Flush();
                 }
+                catch (Exception ex)
+                {
+                    Console.WriteLine(ex.ToString());
```

Comment:
Instead of logging the error in Console, can we log it to Log4Net or somewhere else to review the exception later?

Reply:
The component needs to be modified to allow logger injection. I didn't want to modify the logic so I left this error handling as it was

Reply:
Yes but better we do some changes and log it. Eventually we want to implement Log4net everywhere. Let's do it. Because these are the possible methods crashing the REST API